sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -479,34 +502,46 @@ export class KeysClient {
   /**
    * The delete operation applies to any key stored in Azure Key Vault. Individual versions
    * of a key can not be deleted, only all versions of a given key at once.
+   *
+   * This function returns a Long Running Operation poller that allows you to wait indifinetly until the key is deleted.
+   *
    * This operation requires the keys/delete permission.
    *
    * Example usage:
    * ```ts
-   * let client = new KeysClient(url, credentials);
-   * let result = await client.deleteKey("MyKey");
+   * const client = new KeysClient(url, credentials);
+   * const poller = await client.beginDeleteKey("MyKey");
+   *
+   * // Serializing the poller
+   * const serialized = poller.toJSON();
+   * // A new poller can be created with:
+   * // await client.beginDeleteKey("MyKey", { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const deletedKey = await poller.pollUntilDone();
+   * console.log(deletedKey);
    * ```
    * @summary Deletes a key from a specified key vault.
-   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
    * @param name The name of the key.
    * @param [options] The optional parameters
    */
-  public async deleteKey(name: string, options?: RequestOptions): Promise<DeletedKey> {
-    const requestOptions = (options && options.requestOptions) || {};
-    const span = this.createSpan("deleteKey", requestOptions);
+  public beginDeleteKey(name: string, options?: BeginDeleteKeyOptions): DeleteKeyPoller {
```

Comment:
this method should be `async`

Reply:
I actually wonder if it should be sync instead. Creating the poller doesn't appear to be async. It triggers a non-blocking action but it doesn't appear that the user needs to wait for that non-blocking action to complete before getting the poller.

If the `deleteKey` call triggered by `poller.poll` fails with a terminal state (e.g. Unauthorized), then will `poller.poll` or `poller.pollUntilDone` just return a rejected promise?

Reply:
See this: https://github.com/Azure/azure-sdk-for-js/pull/5575#discussion_r335231048

Reply:
Pollers are created synchronously, but I think Jeff is right that it should be async. `beginDelete` needs to be awaitable so if I don't care to check status, and only care that the request was submitted successfully, I can do `await beginDelete()`.

Unfortunately, this means that the sample happy-path turns into two lines - create the poller, await for it to be done.

Reply:
Gotcha, that makes sense. Thanks for clarifying.

Reply:
Another point to add: we need `begin*` to be consistent across packages so users always know what to expect when they see a begin method. Given that certificates is going with the double-await, we should do so here as well.

Reply:
Changing it!